File: 0242-valid-anagram/0242-valid-anagram.py
# class Solution:
#     def isAnagram(self, s: str, t: str) -> bool:
#         if len(s) != len(t):
#             return False
#         s, t = sorted(s), sorted(t)
#         for i in range(len(s)):
#             if s[i] != t[i]:
#                 return False
#         return True

# class Solution:
#     def isAnagram(self, s: str, t: str) -> bool:
#         temp = []
#         for char in s:
#             if char in temp:
#                 continue
#             if s.count(char) != t.count(char):
#                 return False
#             temp.append(char)
#         return True

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        while (len(s) != 0):
            curr = s[0]
            if s.count(curr) != t.count(curr):
                return False
            s, t = s.replace(curr, ''), t.replace(curr, '')
            if len(s) != len(t):
                return False
        return True

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        // 문자열 길이가 다르면 애너그램이 될 수 없음
        if (s.length() != t.length()) {
            return false;
        }

        while (s.length() > 0) {
            char curr = s.charAt(0); // 첫 번째 문자 가져오기
            long countS = s.chars().filter(c -> c == curr).count(); // s에서 문자 개수 세기
            long countT = t.chars().filter(c -> c == curr).count(); // t에서 문자 개수 세기

            if (countS != countT) {
                return false; // 개수가 다르면 애너그램 아님
            }

            // 해당 문자 제거
            s = s.replace(Character.toString(curr), "");
            t = t.replace(Character.toString(curr), "");

            // 문자열 길이 다시 확인
            if (s.length() != t.length()) {
                return false;
            }
        }

        return true;
    }
}